### PR TITLE
Fix /api/v1/push/subscription

### LINKED ISSUE #11722: POST /api/v1/push/subscription returns status 200 with Html body
- [x] I have searched open and closed issues for duplicates

### Bug Description

The api endpoint POST /api/v1/push/subscription returns status 200 with a html body. This causes client apps like Tusky 19.0 beta 2 to crash as they expect a json response on a success status.

### Steps to Reproduce

1. call POST /api/v1/push/subscription with a valid bearer token and body
2. observe statuscode 200 and a html body being returned

Alternatively you can also try to use Tusky 19.0 beta 2 with Friendica with is currently impossible due to it stopping at startup.

Expected Result:

The push subscription is correctly registered and a [valid json response according to the Mastodon api](https://docs.joinmastodon.org/entities/pushsubscription/) is produced, OR the endpoint fails with a 4xx statuscode.

### Platform Info

Friendica Version: 2022.06 on opensocial.at

(Note, we will fix the crash for Tusky 19.0 but please get this in order on your side as well)

### PR DESCRIPTION
**Pull request by nupplaphil**
Fixes #11722

### CODE REVIEW

File: tests/src/Module/Api/Mastodon/PushSubscriptionTest.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+use Friendica\Test\src\Module\Api\ApiTest;
+
+class PushSubscriptionTest extends ApiTest
+{
+	/**
+	 * Test the api_account_verify_credentials() function.
+	 *
+	 * @return void
+	 */
+	public function testApiAccountVerifyCredentials()
```
Comment (Quix0r) [inline tests/src/Module/Api/Mastodon/PushSubscriptionTest.php]: You already added `: void` in `PushSubscription` class, can you please add it here?

File: tests/src/Module/Api/Mastodon/PushSubscriptionTest.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		//
+		// $this->assertEquals(1,1);
+	}
+
+	/**
+	 * Test the api_account_verify_credentials() function without an authenticated user.
+	 *
+	 * @return void
+	 */
+	public function testApiAccountVerifyCredentialsWithoutAuthenticatedUser()
```
Comment (Quix0r) [inline tests/src/Module/Api/Mastodon/PushSubscriptionTest.php]: You already added `: void` in `PushSubscription` class, can you please add it here?

**Review by Quix0r — CHANGES_REQUESTED** [review]
Otherwise looking fine.

**Review by Quix0r — APPROVED** [review]

File: src/Module/Api/Mastodon/PushSubscription.php [hunk 3]
```diff
@@ -94,37 +112,45 @@ public function put(array $request = [])
 
 		$ret = Subscription::update($application['id'], $uid, $fields);
 
-		Logger::info('Subscription updated', ['result' => $ret, 'application-id' => $application['id'], 'uid' => $uid, 'fields' => $fields]);
+		$this->logger->info('Subscription updated', [
+			'result' => $ret, 'application-id' => $application['id'],
+			'uid'    => $uid, 'fields' => $fields
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/PushSubscription.php]: If you're going for readability, please use the full length declaration. In this context, please add a trailing comma on each line to streamline diffs.
```suggestion
			'result'         => $ret,
			'application-id' => $application['id'],
			'uid'            => $uid,
			'fields'         => $fields,
```

File: src/Module/Api/Mastodon/PushSubscription.php [hunk 4]
```diff
@@ -94,37 +112,45 @@ public function put(array $request = [])
...
 		self::checkAllowedScope(self::SCOPE_PUSH);
 		$uid         = self::getCurrentUserID();
 		$application = self::getCurrentApplication();
 
 		$ret = Subscription::delete($application['id'], $uid);
 
-		Logger::info('Subscription deleted', ['result' => $ret, 'application-id' => $application['id'], 'uid' => $uid]);
+		$this->logger->info('Subscription deleted', [
+			'result' => $ret, 'application-id' => $application['id'],
+			'uid'    => $uid
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/PushSubscription.php]:
```suggestion
			'result'         => $ret,
			'application-id' => $application['id'],
			'uid'            => $uid,
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by Quix0r — APPROVED** [review]

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**